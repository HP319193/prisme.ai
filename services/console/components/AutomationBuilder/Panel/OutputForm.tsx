import { FC, useMemo } from 'react';
import { Schema, SchemaForm } from '@prisme.ai/design-system';
import { useTranslation } from 'next-i18next';
import components from './schemaFormComponents';

interface OutputFormProps {
  output?: string;
  onChange: (v: { output: any }) => void;
}

const buttons: any[] = [];

export const OutputForm: FC<OutputFormProps> = ({ output, onChange }) => {
  const { t } = useTranslation('workspaces');

  const schema: Schema = useMemo(
    () => ({
      additionalProperties: true,
      title: t('automations.output.edit.title'),
      description: t('automations.output.edit.description', {
        interpolation: {
          skipOnVariables: true,
        },
      }),
    }),
    [t]
  );

  return (
    <div className="flex flex-1 flex-col h-full overflow-x-auto">
      <SchemaForm
        schema={schema}
        onChange={onChange}
        initialValues={output}
        components={components}
        buttons={buttons}
      />
    </div>
  );
};

export default OutputForm;
