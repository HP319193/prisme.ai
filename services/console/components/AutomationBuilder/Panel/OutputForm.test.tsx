import OutputForm from './OutputForm';
import renderer from 'react-test-renderer';

jest.mock('../../SchemaForm/BlockSelector', () => {
  return () => null;
});
jest.mock('../../SchemaForm/SchemaForm', () => () => null);

it('should render', () => {
  const onChange = jest.fn();
  const root = renderer.create(
    <OutputForm output="output" onChange={onChange} />
  );
  expect(root.toJSON()).toMatchSnapshot();
});
