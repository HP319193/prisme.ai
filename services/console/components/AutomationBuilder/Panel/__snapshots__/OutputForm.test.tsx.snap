// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="flex flex-1 flex-col h-full overflow-x-auto"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="mt-5"
    >
      automations.output.edit.title
      <div
        className="m-2 mb-5 flex-grow "
      >
        <span
          className="flex flex-col "
        >
          <label
            className="text-gray text-[11px]"
            htmlFor="output"
          >
            automations.output.edit.label
          </label>
          <div />
        </span>
      </div>
    </div>
    automations.output.edit.save
  </form>
</div>
`;
